src/commands/execute.ts

```diff
@@ -0,0 +1,147 @@
+import { Command } from '../types';
+import { Message } from 'discord.js';
+import lodash from 'lodash';
+import * as ts from 'typescript';
+import { VM, CompilerFunction } from 'vm2';
+import { polishPlurals } from 'polish-plurals';
+
+const pluralize = (count: number) => polishPlurals('linia', 'linie', 'linii', count);
+
+/* config */
+export const MaxOutputLines = 20;
+export const MaxOutputCharacters = 1600;
+const Cooldown = 30;
+
+const NodeLanguages: { [key: string]: 'javascript' | CompilerFunction } = {
+  js: 'javascript',
+  ts(code: string): string {
+    const out = ts.transpileModule(code, {
+      compilerOptions: {
+        module: ts.ModuleKind.CommonJS,
+        alwaysStrict: true,
+      },
+    });
+    return out.outputText;
+  },
+};
+
+type ResultType = number | string | object;
+
+export interface ExecuteResult {
+  stdout: string[];
+  result: ResultType;
+  time?: number;
+}
+
+export interface Stdout {
+  text: string;
+  lines: number;
+}
+
+export function parseArg(arg: ResultType) {
+  if (typeof arg === 'object') {
+    return JSON.stringify(arg);
+  } else if (typeof arg !== 'undefined' && arg !== null) {
+    return arg.toString();
+  } else {
+    return arg;
+  }
+}
+
+export function parseMessage(msg: string) {
+  const Delimiter = '```';
+  const opening = msg.indexOf(Delimiter);
+  const closing = msg.lastIndexOf(Delimiter);
+  const newline = msg.indexOf('\n', opening + Delimiter.length);
+  if (opening === -1 || newline === -1) {
```

Comment:
What if `closing` is -1?

Reply:
Right, should check if `closing` is the same as `opening` (if is, throw error). If `opening` is -1, `closing` will be too.